File: MengeCS_bad/Agent.cpp
#include "stdafx.h"

#include "Agent.h"
#include <cassert>
#include <sstream>

namespace MengeCS {

	// ---------------------------------------------------

	Vector2 Agent::GetPosition() {
		assert( _agent != 0x0 );
		const Menge::Math::Vector2 & p = _agent->_pos;
		return Vector2( p._x, p._y );
	}
	
	// ---------------------------------------------------

	Vector2 Agent::GetVelocity() {
		assert( _agent != 0x0 );
		const Menge::Math::Vector2 & p = _agent->_vel;
		return Vector2( p._x, p._y );
	}
	
	// ---------------------------------------------------

	Vector2 Agent::GetOrientation() {
		assert( _agent != 0x0 );
		const Menge::Math::Vector2 & p = _agent->_orient;
		return Vector2( p._x, p._y );
	}
	
	// ---------------------------------------------------

	Vector2 Agent::GetPrefVelocity() {
		assert( _agent != 0x0 );
		const Menge::Math::Vector2 & p = _agent->_velPref.getPreferredVel();
		return Vector2( p._x, p._y );
	}
	
	// ---------------------------------------------------

	size_t Agent::GetClass() {
		assert( _agent != 0x0 );
		return _agent->_class;
	}

	// ---------------------------------------------------

	String^ Agent::ToString() {
		assert( _agent != 0x0 );
		std::stringstream ss;
		ss << "Agent " << _agent->_id;
		return gcnew String( ss.str().c_str() );
	}

	// ---------------------------------------------------

	Agent::Agent( const Menge::Agents::BaseAgent * agent ) : _agent( agent ) {
	}
}

File: MengeCS/Agent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace MengeCS
{
    /// <summary>
    /// Wrapper class for a Menge::Agents::BaseAgent.
    /// </summary>
    public class Agent
    {
        internal Agent()
        {
            _pos = new Vector2();
            _vel = new Vector2();
            _orient = new Vector2();
            _class = 0;
        }

        public Vector2 Position { get { return _pos; } }
        internal Vector2 _pos;

        public Vector2 Velocity { get { return _vel; } }
        internal Vector2 _vel;

        public Vector2 Orientation { get { return _orient; } }
        internal Vector2 _orient;

        public int Class { get { return _class; } }
        internal int _class;
    }
}
